client/dashboard/domains/domain-transfer/transfer-domain-to-other-user.tsx

```diff
@@ -89,15 +103,20 @@ export default function TransferDomainToOtherUser() {
 
 	const onConfirm = () => {
 		const selectedUser = availableUsers.find(
-			( user ) => ( user.id as unknown as string ) === formData.user
+			( user ) => getWpcomUserId( user ) === parseInt( formData.user )
```

Comment:
Nit: I can't imagine the ID in the form data will ever end up with a leading `0` or anything, but I always feel its best practice to pass an explicit radix to `parseInt`

```suggestion
			( user ) => getWpcomUserId( user ) === parseInt( formData.user, 10 )
```